### PR TITLE
React capstone project - Metrics webapp 🎉 🚀 ✔️ 

### PR DESCRIPTION
**Pull request by Nizamuddin4493**
# React capstone project - Metrics webapp 🎉 🚀 ✔️ 

- Use React documentation.
- Use React components.
- Use React props.
- Use React Router.
- Connect React and Redux.
- Handle events in a React app.
- Write integration tests with a React testing library.
- Use styles in a React app.
- Use React life cycle methods.
- Apply React best practices and language style guides in code.
- Use store, actions, and reducers in React.

### CODE REVIEW

File: src/components/HomePage.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <>
+      <div className="homePageDesign">
+        <Header />
+        <hr />
+        <SearchForm />
+        <Routes>
+          <Route path="/" element={<Categories />} />
+          <Route path="/:id" element={<Details />} />
+        </Routes>
+      </div>
```
Comment (mateo951) [inline src/components/HomePage.jsx:19]: The project contains a Home page and a Detail page 🤓 Well done!

File: src/components/HomePage.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Route, Routes } from 'react-router';
```
Comment (mateo951) [inline src/components/HomePage.jsx:2]: You successfully created a single page application (SPA) using react-router-dom to serve each page without reloading 🙌🔥

File: src/components/HomePage.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+function HomePage() {
+  return (
+    <>
+      <div className="homePageDesign">
+        <Header />
+        <hr />
+        <SearchForm />
+        <Routes>
+          <Route path="/" element={<Categories />} />
+          <Route path="/:id" element={<Details />} />
```
Comment (mateo951) [inline src/components/HomePage.jsx:17]: The category item detail page has its unique route within the SPA 🚀

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import { Provider } from 'react-redux';
+import './index.css';
+import App from './App';
+import store from './redux/configureStore';
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+  <React.StrictMode>
+    <BrowserRouter>
+      <Provider store={store}>
```
Comment (mateo951) [inline src/index.js:13]: The project uses Redux for state management and data retrieved from the API are stored in the Redux store correctly 🤓

File: src/__test__/PageRoutes.test.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { Provider } from 'react-redux';
```
Comment (mateo951) [inline src/__test__/PageRoutes.test.js:1]: Your app has unit tests for several components of the app and some pure functions in Redux code! Fantastic job 🔥

File: src/index.css [hunk 6]
```diff
@@ -0,0 +1,164 @@
+* {
```
Comment (mateo951) [inline src/index.css:1]: Please be sure that your app works for mobile sizes 🤓 As the requirements mentioned "mobile web application", having the design work for mobile devices is key 📝

Your design on mobile 
image

**Review by mateo951 — CHANGES_REQUESTED** [review]
### Hi @Nizamuddin4493, 

**🔥Highlights🔥**
- Video presentation ✔
- General Requirements 🤓
- App deployed using Github Pages 🚀
- Implementation of API 🤯

Magnific job so far! Your capstone is looking nice 🔥 You still need to work on some issues to prepare your project for the final evaluation, but you are almost there 😁

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by addod19 — APPROVED** [review]
### Hi @Nizamuddin4493, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Nice job implementing mobile responsive :rocket: 

### Optional
- [ ] It would be better if we capitalize all our commit messages :smiley: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**